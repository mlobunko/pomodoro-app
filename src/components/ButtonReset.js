import React from "react";
import { connect } from "react-redux";
import { resetTimer } from "../actions/saga";

export const ButtonReset = ({ resetTimer }) => (
  <div>
    <button
      aria-label="reset button"
      className="button button-side"
      onClick={resetTimer}
    >
      <i className="fa fa-refresh" aria-hidden="true" />
    </button>
  </div>
);

export default connect(undefined, { resetTimer })(ButtonReset);
