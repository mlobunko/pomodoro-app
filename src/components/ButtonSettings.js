import React from "react";
import { connect } from "react-redux";
import { openSettings } from "../actions/statistics";

export const ButtonSettings = ({ openSettings }) => (
  <div>
    <button
      aria-label="settings button"
      className="button button-side"
      onClick={openSettings}
    >
      <i className="fa fa-cog" />
    </button>
  </div>
);

export default connect(undefined, { openSettings })(ButtonSettings);
