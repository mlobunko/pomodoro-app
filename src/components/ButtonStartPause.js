import React from "react";
import { connect } from "react-redux";
import { startTimer } from "../actions/saga";
import { timerOnToFalse } from "../actions/statistics";

export const ButtonStartPause = ({ timerOn, startTimer, timerOnToFalse }) => (
  <div>
    {timerOn ? (
      <button className="button" onClick={timerOnToFalse}>
        <i class="fa fa-pause" />
      </button>
    ) : (
      <button className="button" onClick={startTimer}>
        <i className="fa fa-play" />
      </button>
    )}
  </div>
);

const mapStateToProps = state => ({
  timerOn: state.statistics.timerOn
});

export default connect(mapStateToProps, { startTimer, timerOnToFalse })(
  ButtonStartPause
);
