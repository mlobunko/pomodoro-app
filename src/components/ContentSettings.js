import React from "react";
import { connect } from "react-redux";
import {
  increaseTimePeriodWork,
  decreaseTimePeriodWork,
  increaseTimePeriodRest,
  decreaseTimePeriodRest,
  increaseTimePeriodBigRest,
  decreaseTimePeriodBigRest,
  changeIsSound
} from "../actions/settings";
import { displaySettings } from "../logic/convert";
import { closeSettings } from "../actions/statistics";

export const ContentSettings = ({
  timePeriodWork,
  timePeriodRest,
  timePeriodBigRest,
  isSound,
  closeSettings,
  increaseTimePeriodWork,
  decreaseTimePeriodWork,
  increaseTimePeriodRest,
  decreaseTimePeriodRest,
  increaseTimePeriodBigRest,
  decreaseTimePeriodBigRest,
  changeIsSound
}) => (
  <div className="content-settings">
    <div className="header">
      <h1>Settings</h1>
    </div>
    <div className="container">
      <div className="settings-content">
        <div>Work</div>
        <div>
          <button
            onClick={decreaseTimePeriodWork}
            className="button-plus-minus"
          >
            <i class="fa fa-minus" />
          </button>
          {displaySettings(timePeriodWork)}
          <button
            onClick={increaseTimePeriodWork}
            className="button-plus-minus"
          >
            <i class="fa fa-plus" />
          </button>
        </div>
        <div>Rest</div>
        <div>
          <button
            onClick={decreaseTimePeriodRest}
            className="button-plus-minus"
          >
            <i class="fa fa-minus" />
          </button>
          {displaySettings(timePeriodRest)}
          <button
            onClick={increaseTimePeriodRest}
            className="button-plus-minus"
          >
            <i class="fa fa-plus" />
          </button>
        </div>
        <div>Big rest</div>
        <div>
          <button
            onClick={decreaseTimePeriodBigRest}
            className="button-plus-minus"
          >
            <i class="fa fa-minus" />
          </button>
          {displaySettings(timePeriodBigRest)}
          <button
            onClick={increaseTimePeriodBigRest}
            className="button-plus-minus"
          >
            <i class="fa fa-plus" />
          </button>
        </div>
        <div>Sound</div>
        <div onClick={changeIsSound} className="button-plus-minus">
          {isSound ? (
            <i class="fa fa-volume-up" />
          ) : (
            <i class="fa fa-volume-down" />
          )}
        </div>
      </div>
      <div className="attribute">
        Icons made by Freepik from www.flaticon.com
      </div>
    </div>
    <div className="buttons">
      <div className="button-settings-close">
        <button onClick={closeSettings} className="button">
          OK
        </button>
      </div>
    </div>
  </div>
);

const mapStateToProps = state => ({
  timePeriodWork: state.settings.timePeriodWork,
  timePeriodRest: state.settings.timePeriodRest,
  timePeriodBigRest: state.settings.timePeriodBigRest,
  isSound: state.settings.isSound
});

export default connect(mapStateToProps, {
  closeSettings,
  increaseTimePeriodWork,
  decreaseTimePeriodWork,
  increaseTimePeriodRest,
  decreaseTimePeriodRest,
  increaseTimePeriodBigRest,
  decreaseTimePeriodBigRest,
  changeIsSound
})(ContentSettings);
