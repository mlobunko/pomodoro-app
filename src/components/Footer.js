import React from "react";

const Footer = () => (
  <div className="footer">
    <span>Icon made by Freepik from www.flaticon.com</span>
  </div>
);

export default Footer;
