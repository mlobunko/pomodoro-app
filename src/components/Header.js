import React from "react";

const Header = () => (
  <div className="header">
    <h1>
      P<div className="little-tomato" />modoro Timer
    </h1>
    <a
      href="https://en.wikipedia.org/wiki/Pomodoro_Technique"
      target="_blank"
      rel="noopener noreferrer"
    >
      <div>
        <i className="fa fa-info-circle" />
      </div>
    </a>
  </div>
);

export default Header;
