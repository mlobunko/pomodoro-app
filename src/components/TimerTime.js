import React from "react";
import { connect } from "react-redux";
import { msToMmSs } from "../logic/convert";

export const TimerTime = ({ isWorkingTime, displayTimer }) => (
  <div className={isWorkingTime ? "timerTime" : "timerTime timerTime--rest"}>
    {msToMmSs(displayTimer)}
  </div>
);

const mapStateToProps = ({ statistics }) => ({
  isWorkingTime: statistics.isWorkingTime,
  displayTimer: statistics.displayTimer
});

export default connect(mapStateToProps)(TimerTime);
