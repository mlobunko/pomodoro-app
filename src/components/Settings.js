import React from "react";
import ReactModal from "react-modal";
import { connect } from "react-redux";
import ContentSettings from "./ContentSettings";
import { closeSettings } from "../actions/statistics";

export const Settings = ({ isSettingsOpen }) => (
  <div>
    <ReactModal
      isOpen={isSettingsOpen}
      onRequestClose={closeSettings}
      contentLabel="Settings"
      className="Modal"
      overlayClassName="Overlay"
      ariaHideApp={false}
    >
      <ContentSettings />
    </ReactModal>
  </div>
);

const mapStateToProps = state => ({
  isSettingsOpen: state.statistics.isSettingsOpen
});

export default connect(mapStateToProps, { closeSettings })(Settings);
