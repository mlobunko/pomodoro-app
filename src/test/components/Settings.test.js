import React from "react";
import { shallow, mount } from "enzyme";
import { Settings } from "../../components/Settings";
import { CustomProvider } from "./CustomProvider";

it("should render Settings component correctly", () => {
  const wrapper = shallow(<Settings />);
  expect(wrapper).toMatchSnapshot();
});

describe("modal showing", () => {
  it("show modal when isOpen is true", () => {
    const props = {
      isSettingsOpen: true
    };
    const wrapper = mount(
      <CustomProvider>
        <Settings {...props} />
      </CustomProvider>
    );
    expect(wrapper.find(".Modal").exists()).toEqual(true);
    expect(wrapper.find(".content-settings").exists()).toEqual(true);
  });

  it("hide modal when isOpen is false", () => {
    const props = {
      isSettingsOpen: false
    };
    const wrapper = mount(
      <CustomProvider>
        <Settings {...props} />
      </CustomProvider>
    );
    expect(wrapper.find(".Modal").exists()).toEqual(true);
    expect(wrapper.find(".content-settings").exists()).toEqual(false);
  });
});
