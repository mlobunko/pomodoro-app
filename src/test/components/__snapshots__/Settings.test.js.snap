// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Settings component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Modal
        ariaHideApp={false}
        bodyOpenClassName="ReactModal__Body--open"
        className="Modal"
        closeTimeoutMS={0}
        contentLabel="Settings"
        isOpen={false}
        onRequestClose={[Function]}
        overlayClassName="Overlay"
        parentSelector={[Function]}
        portalClassName="ReactModalPortal"
        shouldCloseOnEsc={true}
        shouldCloseOnOverlayClick={true}
        shouldFocusAfterRender={true}
        shouldReturnFocusAfterClose={true}
>
        <Connect(Component) />
</Modal>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "ariaHideApp": false,
        "bodyOpenClassName": "ReactModal__Body--open",
        "children": <Connect(Component) />,
        "className": "Modal",
        "closeTimeoutMS": 0,
        "contentLabel": "Settings",
        "isOpen": false,
        "onRequestClose": [Function],
        "overlayClassName": "Overlay",
        "parentSelector": [Function],
        "portalClassName": "ReactModalPortal",
        "shouldCloseOnEsc": true,
        "shouldCloseOnOverlayClick": true,
        "shouldFocusAfterRender": true,
        "shouldReturnFocusAfterClose": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Modal
          ariaHideApp={false}
          bodyOpenClassName="ReactModal__Body--open"
          className="Modal"
          closeTimeoutMS={0}
          contentLabel="Settings"
          isOpen={false}
          onRequestClose={[Function]}
          overlayClassName="Overlay"
          parentSelector={[Function]}
          portalClassName="ReactModalPortal"
          shouldCloseOnEsc={true}
          shouldCloseOnOverlayClick={true}
          shouldFocusAfterRender={true}
          shouldReturnFocusAfterClose={true}
>
          <Connect(Component) />
</Modal>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "ariaHideApp": false,
          "bodyOpenClassName": "ReactModal__Body--open",
          "children": <Connect(Component) />,
          "className": "Modal",
          "closeTimeoutMS": 0,
          "contentLabel": "Settings",
          "isOpen": false,
          "onRequestClose": [Function],
          "overlayClassName": "Overlay",
          "parentSelector": [Function],
          "portalClassName": "ReactModalPortal",
          "shouldCloseOnEsc": true,
          "shouldCloseOnOverlayClick": true,
          "shouldFocusAfterRender": true,
          "shouldReturnFocusAfterClose": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
